### PR TITLE
feat(comit) `XMR<>BTC` integration

### PR DESCRIPTION
**Pull request by da-kami**

### CODE REVIEW

**Comment by da-kami** [comment]
At the moment fetching the quote on `localhost` works, but once deployed we run into problems. 

@thomaseizinger will we need a certificate + WSS connection over libp2p to get this working? 
Would be great to get some input, I don't know how to solve the issues in production.

**Comment by da-kami** [comment]
Somehow I need https://github.com/opendexnetwork/opendex.app/pull/36/commits/3ec6512d8dfe2a660fac8c322eab39e6b39854b0 for getting prettier fixed. Can someone explain to me why? I rebased on top of `main` which has passing prettier - where are these changes coming from? 😅
As far as I can see https://github.com/opendexnetwork/opendex.app/pull/36/commits/26ce55f062b1adb912558db73ee3355a2b795cb3 does not touch those files...

**Comment by thomaseizinger** [comment]
> I rebased on top of `main` which has passing prettier - where are these changes coming from? sweat_smile

Are you rebasing on top of `upstream/main` or your local main? With forks, it can be tricky to get these things right. If that isn't it, I don't know what is wrong :(

File: src/components/ComitSwapFlow/Asb.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { BitcoinAmount } from './BitcoinAmount';
+import { Network } from '../../context/NetworkContext';
+import Decimal from 'decimal.js';
+
+// At the moment the
```
Comment (thomaseizinger) [inline src/components/ComitSwapFlow/Asb.ts:5]: Only half a sentence :)

File: src/components/DrawQrCode/index.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ class QrCode extends React.Component {
 
 QrCode.propTypes = {
   size: PropTypes.number.isRequired,
-  link: PropTypes.string.isRequired,
+  step_text: PropTypes.string.isRequired,
```
Comment (thomaseizinger) [inline src/components/DrawQrCode/index.js:52]: Is this change intended?

File: src/components/ComitSwapFlow/XmrBtcRateFetcher.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  previewGivenReceive(
+    amountWithCurrency: CurrencyAmount,
+    pair: CurrencyPair
+  ): Promise<AmountPreview> {
+    this.ensureNoRefreshErrors();
+    let receive_xmr = MoneroAmount.fromXmr(
+      amountWithCurrency.amount.toString()
+    );
+
+    console.log(receive_xmr.asXmr().toString());
```
Comment (thomaseizinger) [inline src/components/ComitSwapFlow/XmrBtcRateFetcher.ts:44]: Should this stay?

**Review by thomaseizinger — APPROVED** [review]

File: src/components/DrawQrCode/index.js (hunk 2 above)
Comment (da-kami) [reply src/components/DrawQrCode/index.js:52]: not sure how that slipped in there, thanks :)

File: src/components/ComitSwapFlow/Asb.ts (hunk 1 above)
Comment (da-kami) [reply src/components/ComitSwapFlow/Asb.ts:5]: yeah, I thought about adding some kind of disclaimer there that explains why we don't actually fetch the rate from the ASB, but I don't think it adds that much value in a comment. Removed :)
